elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -148,8 +151,8 @@ def __trace(method, path, params, body, url, response, json, took, duration)
         #
         # @api private
         def __raise_transport_error(response)
-          error = ERRORS[response.status] || ServerError
-          raise error.new "[#{response.status}] #{response.body}"
+          error = ERRORS[response.status].new(response) || ServerError.new(response)
```

Comment:
any reason to move the creation of the exception to this line instead of line 155? no strong preference but

```
error_class = ERRORS.fetch(response.status, ServerError)
raise error_class.new(response), "[#{response.status}] #{response.body}"
```
seems more readable to me